### DISCUSSION TITLE
product key verification failed

### DISCUSSION THREAD

**Discussion by yayaasd** [zsrv/supermicro-product-key#10] [General]
I am trying to generate a license, but always got a "Error: product key verification failed: crypto/rsa: verification error" during the step:

`./supermicro-product-key json verify MAC '{"ProductKey":{"Node":{"LicenseID":"2","LicenseName":"SFT-DCMS-SINGLE","CreateDate":"20240221"},"Signature":"SIGNATURE-KEY"}}'`

any idea?

**Comment by yayaasd** [comment]
oh and btw - installing via API also results in:

`{
    "error": {
        "code": "Base.v1_10_3.GeneralError",
        "message": "A general error has occurred. See ExtendedInfo for more information.",
        "@Message.ExtendedInfo": [
            {
                "MessageId": "SMC.1.0.OemActivatelicenseErrorMessage",
                "Severity": "Warning",
                "Resolution": "Please check the reason for the error.",
                "Message": "Error: License key input is wrong..",
                "MessageArgs": [
                    "License key input is wrong."
                ],
                "RelatedProperties": [
                    "ActivateLicenseErrors"
                ]
            }
        ]
    }
}`




input -->
`{"ProductKey":{"Node":{"LicenseID":"2","LicenseName":"SFT-DCMS-SINGLE","CreateDate":"20240221"},"Signature":"SIGNATURE-KEY"}}`

**Comment by zsrv** [comment]
Has any part of the product key been altered? If any part of the text was changed, the signature verification will fail.

Another possibility is that the license is actually associated with a different MAC address. Try using the bruteforce function in the product key utility to see if the key can be matched to a different MAC address in one of the Supermicro MAC address ranges.

**Reply by yayaasd** [reply]
nope, nothing...
i also did the bruteforcing and after minutes I was able to get the input MAC as the result..

**Reply by zsrv** [reply]
Which motherboard is the MAC address associated with? Which BMC firmware version are you using?

**Reply by yayaasd** [reply]
this is H12SSL-CT with BMC Firmware 01.04.03

**Reply by zsrv** [reply]
Okay, I'm not sure then. The license signing key was not changed in that firmware version, so if the "json verify" command fails then that means the MAC is different or something in the license was altered.
